src/rust/engine/src/context.rs

```diff
@@ -219,6 +220,22 @@ impl Core {
             exec_strategy_opts.local_keep_sandboxes,
         );
 
+        // TODO: Wrap this in a BoundedCommandRunner so only run process can execute in the workspace at once.
```

Comment:
Thoughts on using `BoundedCommandRunner` to enforce this?

Reply:
Are the sandboxes uniquely named such that they wouldn't collide?

For cases where we are invoking an existing tool, we'd expect them to do the right thing with regard to locking so that they could safely run concurrently. But I wonder whether it would make it challenging to write correct end users scripts to have concurrency.

It sounds like mutability of declared outputs is hidden, because those go into the sandbox? So only undeclared outputs that didn't go into temporary directories would be a problem.

I think it's fine to ship without limiting to 1... in future could add a concurrency limit as a workspace option perhaps...?